examples/bq-email-exports/main.py

```diff
@@ -0,0 +1,181 @@
+    # with path to service account json key file
+    else:
+        credentials = service_account.Credentials.from_service_account_file(
+            os.getenv("GOOGLE_APPLICATION_CREDENTIALS"),
+            scopes=["https://www.googleapis.com/auth/cloud-platform"],
+        )
+    return credentials
+
+
+def main():
+    """Sends an email with a Google Cloud Storage signed URL of BQ Query results.
+    Creates BQ table, runs a SQL query, and exports the results to Cloud Storage as a CSV.
+    Generates signing credentials for the CSV and sends an email with a link to the signed URL.
+
+    Args:
+        None
+
+    Returns:
+        None
+    """
+
+    # Create BQ and Storage Client
+    bq_client = bigquery.Client(credentials=credentials())
+    storage_client = storage.Client(credentials=credentials())
+
+    # Set variables
+    timestr = time.strftime("%Y%m%d%I%M%S")
+    project = "report-scheduling"
+    dataset_id = "bq_exports"
+    file_name = "daily_export_" + timestr
+    csv_name = file_name + ".csv"
+    table_id = "report-scheduling.bq_exports." + file_name
+    bucket_name = "bq_email_exports"
+    from_email = "sender@example.com"
+    to_emails = "recipient@example.com"
+
+    # Create a BQ table
+    schema = [
+        bigquery.SchemaField("url", "STRING", mode="REQUIRED"),
+        bigquery.SchemaField("view_count", "INTEGER", mode="REQUIRED"),
+    ]
+
+    # Make an API request to create table
+    table = bq_client.create_table(bigquery.Table(table_id, schema=schema))
+    print("Created table {}.{}.{}".format(table.project, table.dataset_id,
+                                          table.table_id))
+
+    # Run query on that table
+    job_config = bigquery.QueryJobConfig(destination=table_id)
+
+    # Define the SQL query
+    sql = """
+        SELECT
+        CONCAT(
+            'https://stackoverflow.com/questions/',
+            CAST(id as STRING)) as url,
+        view_count
+        FROM `bigquery-public-data.stackoverflow.posts_questions`
+        WHERE tags like '%google-bigquery%'
+        ORDER BY view_count DESC
+        LIMIT 10
+    """
+
+    # Start the query, passing in the extra configuration
+    query_job = bq_client.query(sql, job_config=job_config)
+
+    # Wait for the job to complete
+    query_job.result()
+    print("Query results loaded to the table {}".format(table_id))
+
+    # Export table data as CSV to GCS
+    destination_uri = "gs://{}/{}".format(bucket_name, csv_name)
+    dataset_ref = bq_client.dataset(dataset_id, project=project)
+    table_ref = dataset_ref.table(file_name)
+
+    extract_job = bq_client.extract_table(
+        table_ref,
+        destination_uri,
+        # Location must match that of the source table
+        location="US",
+    )
+
+    # Waits for job to complete
+    extract_job.result()
+    print("Exported {}:{}.{} to {}".format(project, dataset_id, table_id,
+                                           destination_uri))
+
```

Comment:
At this point shouldn't you clean up the temporary BQ table?

Reply:
Because this is such a short lived table, should we just leverage the anonymous table which will get cleaned up for us? (and is "free") @tswast thoughts?
https://cloud.google.com/bigquery/docs/cached-results

Reply:
I cleaned up the table for now, but open to trying the anonymous table too

Reply:
bump @tswast do you think this is an appropriate use of anonymous table (because it's free and less for us to manage here)?